app/src/main/java/com/example/triptracker/viewmodel/HomeViewModel.kt

```diff
@@ -207,4 +203,71 @@ class HomeViewModel(private val repository: ItineraryRepository = ItineraryRepos
   fun setSearchFilter(filterType: FilterType) {
     _selectedFilter.value = filterType
   }
+
+  /**
+   * Filter the itinerary list by trending itineraries based on the flame count The trending
+   * itineraries are sorted in descending order of flame count
+   */
+  fun filterByTrending() {
+    _itineraryList.value =
+        _itineraryList.value?.sortedByDescending { itinerary -> itinerary.flameCount }
+  }
+
+  private fun calculateFlameCounts(saves: Long, clicks: Long, numStarts: Long): Long {
+    return 2 * saves + clicks + 5 * numStarts
+  }
```

Comment:
Nice functions !

Reply:
Maybe in the comment of the filterByTrending() we might explain that the filtered list is then stored in _itineraryList ?